hikari/impl/stateful_cache.py

```diff
@@ -111,9 +114,11 @@ class StatefulCacheImpl(cache.MutableCache):
         cache_utility.GenericRefWrapper[emojis.CustomEmoji],
     ]
     _user_entries: collections.ExtendedMutableMapping[snowflakes.Snowflake, cache_utility.GenericRefWrapper[users.User]]
+    _message_entries: collections.ExtendedMutableMapping[snowflakes.Snowflake, messages.PartialMessage]
     _intents: intents_.Intents
+    max_messages: int
 
-    def __init__(self, app: traits.RESTAware, intents: intents_.Intents) -> None:
+    def __init__(self, app: traits.RESTAware, intents: intents_.Intents, max_messages: int) -> None:
```

Comment:
This might not be the best way to handle configuration for the cache as in the future there will be more configuration such as all resources being individually togglable and the unlikely but still to be considered approach to configuration the caching approach for specific resources. Might want to make a dataclass for the cache config preliminarily but i'm still thinking over the best approach for cache config personally